[Dialog] Fix broken styles if `maxWidth` is set to `false`

kmurgic opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Fixes #32833 

**Before:** https://codesandbox.io/s/simpledialog-material-demo-forked-u2nl0u

**Now:** https://codesandbox.io/s/simpledialog-material-demo-forked-zzmpxr

ZeeshanTamboli commented:
Can you add tests?
1. Check if `MuiDialog-paperWidthFalse` class gets applied when `maxWidth` is `false`.
2. Ensure correct `max-width` style is applied when maxWidth is false.

I can help you if needed.

kmurgic commented:
@ZeeshanTamboli  this ci check is failing, but seems to me like it is possibly not related to my PR? Can you confirm if this ci failure is related to my changes or not: https://github.com/mui/material-ui/runs/6702737396?check_suite_focus=true

ZeeshanTamboli commented:
> @ZeeshanTamboli this ci check is failing, but seems to me like it is possibly not related to my PR? Can you confirm if this ci failure is related to my changes or not: https://github.com/mui/material-ui/runs/6702737396?check_suite_focus=true

It's not related to your changes. Reran the CI and now it passed. It was flaky I guess.

kmurgic commented:
> Change looks good to me. Can you add tests?
> 
>     1. Check if `MuiDialog-paperWidthFalse` class gets applied when `maxWidth` is `false`.
> 
>     2. Ensure correct `max-width` style is applied when maxWidth is false.
> 
> 
> I can help you if needed.

I added tests that the paperWidthFalse class is applied when maxWidth={false} but wasn't quite sure what you wanted for testing that the correct styles get applied when max-width is false, since from what I can tell no styles should be applied and I have no idea how I would test that the broken styles aren't getting applied.

If the test I added is not sufficient please let me know specifically what test I should add and I could probably use a little guidance in writing it. Thanks!!

ZeeshanTamboli commented:
> > Change looks good to me. Can you add tests?
> > ```
> > 1. Check if `MuiDialog-paperWidthFalse` class gets applied when `maxWidth` is `false`.
> > 
> > 2. Ensure correct `max-width` style is applied when maxWidth is false.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I can help you if needed.
> 
> I added tests that the paperWidthFalse class is applied when maxWidth={false} but wasn't quite sure what you wanted for testing that the correct styles get applied when max-width is false, since from what I can tell no styles should be applied and I have no idea how I would test that the broken styles aren't getting applied.
> 
> If the test I added is not sufficient please let me know specifically what test I should add and I could probably use a little guidance in writing it. Thanks!!

The first test that you added is correct. 👍 Thanks.

For the second test, you need to check that the broken style `max-width: undefinedpx` style do not get applied. For that you can simple check that correct style `max-width: calc(100% - 64px)` gets applied instead. 

[Here](https://github.com/mui/material-ui/blob/170c3cafd1d51757724f58cacd0f92e03ce0974e/packages/mui-material/src/Dialog/Dialog.test.js#L319-L330) is a test reference on how you can add it by asserting on styles. Let me know if it helps.

kmurgic commented:
Added a test for the styles here: d755d2af22b4c4a5cbdf4030f7bbd651f42f71b7

I'm not sure how useful that test is as I confirmed my suspicion that the test passes with or without my changes since the `max-width: undefinedpx` style is invalid and doesn't actually get applied.

kmurgic commented:
Not sure why visual tests that were passing are now failing after changes to a test file

ZeeshanTamboli reviewed (approved):
It's a great first pull request on Material UI 👌 Thank you for working on it!
